### PR TITLE
[13.0][IMP] account_invoice_margin_sale_delivered_sync: Add option setting to non update delivered margin on posted invoices

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT37653

ping @carlosdauden @CarlosRoca13

### CODE REVIEW

File: account_invoice_margin_sale_delivered_sync/models/account_move.py [hunk 1]
```diff
@@ -19,4 +19,15 @@ def _get_purchase_price(self):
         "product_id", "product_uom_id", "sale_line_ids.purchase_price_delivery"
     )
     def _compute_purchase_price(self):
-        return super()._compute_purchase_price()
+        """
+        Exclude posted invoice lines depending on company setting values
+        """
```
Comment (pedrobaeza) [inline account_invoice_margin_sale_delivered_sync/models/account_move.py]:
```suggestion
        """Exclude posted invoice lines depending on company setting values."""
```

File: account_invoice_margin_sale_delivered_sync/models/account_move.py [hunk 2]
```diff
@@ -19,4 +19,15 @@ def _get_purchase_price(self):
...
+        lines_posted = self.browse()
+        for company in self.mapped("company_id"):
+            if not company.margin_sale_sync_invoice_posted:
+                lines_posted += self.filtered(
+                    lambda ln: (
+                        ln.company_id == company and ln.parent_state == "posted"
+                    )
+                )
```
Comment (pedrobaeza) [inline account_invoice_margin_sale_delivered_sync/models/account_move.py]:
```suggestion
        lines_posted = self.filtered(lambda x: not x.company_id.margin_sale_sync_invoice_posted and ln.parent_state == 'posted')
```

File: account_invoice_margin_sale_delivered_sync/i18n/es.po [hunk 3]
```diff
@@ -6,28 +6,45 @@ msgid ""
...
 msgid "Sales Order Line"
 msgstr "Línea de pedido de venta"
+
+#. module: account_invoice_margin_sale_delivered_sync
+#: model:ir.model.fields,field_description:account_invoice_margin_sale_delivered_sync.field_res_company__margin_sale_sync_invoice_posted
+#: model:ir.model.fields,field_description:account_invoice_margin_sale_delivered_sync.field_res_config_settings__margin_sale_sync_invoice_posted
+msgid "Sync sale delivered margin on posted invoices"
+msgstr "Sincronizar margen de lo entregado en facturas validadas"
+
+#~ msgid "Invoice Line"
```
Comment (pedrobaeza) [inline account_invoice_margin_sale_delivered_sync/i18n/es.po]: You can purge this

File: account_invoice_margin_sale_delivered_sync/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    margin_sale_sync_invoice_posted = fields.Boolean(
```
Comment (cesar-tecnativa) [inline account_invoice_margin_sale_delivered_sync/models/res_company.py:9]: company_dependent=True

File: account_invoice_margin_sale_delivered_sync/models/res_config_settings.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    margin_sale_sync_invoice_posted = fields.Boolean(
```
Comment (cesar-tecnativa) [inline account_invoice_margin_sale_delivered_sync/models/res_config_settings.py:9]: company_dependent=True

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

File: account_invoice_margin_sale_delivered_sync/models/res_company.py (hunk 4 above)
Comment (pedrobaeza) [reply account_invoice_margin_sale_delivered_sync/models/res_company.py:9]: The field is in the company, so there's no sense to make it company dependent

Comment (cesar-tecnativa) [reply account_invoice_margin_sale_delivered_sync/models/res_company.py:9]: you are right

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!!

@cesar-tecnativa clarified about the company!!

**Review by pedrobaeza — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**